asserts/ifacedecls_test.go

```diff
@@ -266,5 +295,231 @@ foo: [{p: "/foo/x"}, {p: "/foo/y"}]
+		},
+	})
+	c.Assert(err, IsNil)
+
+	cstrs := rule.AllowConnection
+	c.Check(cstrs.SlotSnapTypes, DeepEquals, []string{"core", "kernel", "gadget", "app"})
+	c.Check(cstrs.SlotSnapIDs, DeepEquals, []string{"snapidsnapidsnapidsnapidsnapid01", "snapidsnapidsnapidsnapidsnapid02"})
+	c.Check(cstrs.SlotPublisherIDs, DeepEquals, []string{"pubidpubidpubidpubidpubidpubid09", "canonical", "$same"})
+
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsAttributesDefault(c *C) {
+	rule, err := asserts.CompilePlugRule("iface", map[string]interface{}{
+		"allow-connection": map[string]interface{}{
+			"slot-snap-id": []interface{}{"snapidsnapidsnapidsnapidsnapid01"},
+		},
+	})
+	c.Assert(err, IsNil)
+
+	// attributes default to always matching here
+	cstrs := rule.AllowConnection
+	c.Check(cstrs.PlugAttributes, Equals, asserts.AlwaysMatchAttributes)
+	c.Check(cstrs.SlotAttributes, Equals, asserts.AlwaysMatchAttributes)
+}
+
+func (s *plugSlotRulesSuite) TestCompilePlugRuleErrors(c *C) {
+	tests := []struct {
+		stanza string
+		err    string
+	}{
+		{`iface: foo`, `plug rule for interface "iface" must be a map or one of the shortcuts 'true' or 'false'`},
+		{`iface:
+  - allow`, `plug rule for interface "iface" must be a map or one of the shortcuts 'true' or 'false'`},
+		{`iface:
+  allow-installation: foo`, `allow-installation in plug rule for interface "iface" must be a map or one of the shortcuts 'true' or 'false'`},
+		{`iface:
+  allow-connection: foo`, `allow-connection in plug rule for interface "iface" must be a map or one of the shortcuts 'true' or 'false'`},
+		{`iface:
+  allow-installation:
+    plug-attributes:
+      a1: [`, `cannot compile plug-attributes in allow-installation in plug rule for interface "iface": cannot compile "a1" constraint .*`},
+		{`iface:
+  allow-connection:
+    slot-attributes:
+      a2: [`, `cannot compile slot-attributes in allow-connection in plug rule for interface "iface": cannot compile "a2" constraint .*`},
+		{`iface:
+  allow-connection:
+    slot-snap-id:
+      -
+        foo: 1`, `slot-snap-id in allow-connection in plug rule for interface "iface" must be a list of strings`},
+		{`iface:
+  allow-connection:
+    slot-snap-id:
+      - foo`, `slot-snap-id in allow-connection in plug rule for interface "iface" contains an invalid element: "foo"`},
+		{`iface:
+  allow-connection:
+    slot-snap-type:
+      - foo`, `slot-snap-type in allow-connection in plug rule for interface "iface" contains an invalid element: "foo"`},
+		{`iface:
+  allow-connection:
+    slot-snap-type:
+      - xapp`, `slot-snap-type in allow-connection in plug rule for interface "iface" contains an invalid element: "xapp"`},
+		{`iface:
+  allow-connection:
+    slot-snap-ids:
+      - foo`, `allow-connection in plug rule for interface "iface" must specify at least one of plug-attributes, slot-attributes, slot-snap-type, slot-publisher-id, slot-snap-id`},
+		{`iface:
```

Comment:
Can you add a deny-connection, allow-auto-connection and deny-auto-connection test? Not saying we have to do all the tests that you did for allow-connection here for each these, but just a test for each to help future-proof would be nice.